vcGL/src/vcTextureHelper.cpp

```diff
@@ -246,6 +246,77 @@ udResult vcTexture_ResizePixels(const void *pPixels, uint32_t width, uint32_t he
   return result;
 }
 
+udResult vcTexture_GetPixelSize(vcTextureFormat format, size_t *pSize)
+{
+  static const int ByteCount[vcTextureFormat_Count] = {-1, 4, 8, 16, 2, 4};
+
+  if (format == vcTextureFormat_Count || pSize == nullptr)
+    return udR_InvalidParameter_;
+
+  *pSize = ByteCount[format];
+  return udR_Success;
+}
+
+// TODO: This is duplicated from vaultClient: vcRender.cpp, which is to be moved to udCore (AB#1573).
+static float Float16ToFloat32(uint16_t float16)
+{
+  uint16_t sign_bit = (float16 & 0b1000000000000000) >> 15;
+  uint16_t exponent = (float16 & 0b0111110000000000) >> 10;
+  uint16_t fraction = (float16 & 0b0000001111111111) >> 0;
+
+  float sign = (sign_bit) ? -1.0f : 1.0f;
+  float m = 1.0f;
+  uint16_t exponentBias = 15;
+
+  // The exponents '00000' and '11111' are interpreted specially
+  if (exponent == 31)
+    return sign * INFINITY;
+
+  if (exponent == 0)
+  {
+    m = 0.0f;
+    exponentBias = 14;
+  }
+
+  return sign * udPow(2.0f, float(exponent - exponentBias)) * (m + (fraction / 1024.0f));
+}
+
+udResult vcTexture_ConvertPixels(const void *pPixels, vcTextureFormat formatIn, uint8_t **ppPixelsOut, vcTextureFormat formatOut, size_t pixelCount)
+{
+  udResult result = udR_Failure_;
+
+  UD_ERROR_IF((pPixels == nullptr) || (ppPixelsOut == nullptr) || (pixelCount == 0) || (formatIn == vcTextureFormat_Unknown) || (formatIn == vcTextureFormat_Count) || (formatOut == vcTextureFormat_Unknown) || (formatOut == vcTextureFormat_Count), udR_Failure_);
+
+  if (formatIn == formatOut)
+  {
+    size_t formatSize;
+    UD_ERROR_CHECK(vcTexture_GetPixelSize(formatIn, &formatSize));
+    *ppPixelsOut = udAllocType(uint8_t, pixelCount * formatSize, udAF_Zero);
+    UD_ERROR_NULL(ppPixelsOut, udR_MemoryAllocationFailure);
+    memcpy(*ppPixelsOut, pPixels, pixelCount * formatSize);
+  }
+  else if (formatIn == vcTextureFormat_RGBA16F && formatOut == vcTextureFormat_RGBA8)
+  {
+    *ppPixelsOut = udAllocType(uint8_t, pixelCount * 4, udAF_Zero);
+    UD_ERROR_NULL(ppPixelsOut, udR_MemoryAllocationFailure);
+    for (size_t i = 0; i < pixelCount; ++i)
+    {
+      for (int e = 0; e < 4; ++e)
+      {
+        float temp = Float16ToFloat32(((uint16_t*)pPixels)[i * 4 + e]);
```

Comment:
This could be < 0.0f, and > 1.0f. IMO just clamp:
`temp = udClamp(temp, 0.0f, 1.0f);`